docs/guides.md

```diff
@@ -523,3 +523,79 @@ Now we enable autostart at the virtual machine level.
 5. Checking the output 
 `# xe vm-param-list uuid=<VM_UUID> | grep other-config`
 
+## UEFI Secure Boot
+
+UEFI Secure Boot is a mechanism that allows platform owners to trust that their platforms are only executing system software binaries that they trust and pre-approve.
+
+XCP-ng supports UEFI Secure Boot for guests and is undergoing active development to enable UEFI Secure Boot for hosts.
+
+### Guest UEFI Secure Boot
+
+Enabling UEFI Secure Boot for guests ensures that XCP-ng will ensure that guest VMs only execute trusted binaries.  In practice, these are the binaries released by the operating system (OS) team for the OS in running in the VM (Microsoft Windows, Debian, Alpine, etc.).
+
+#### Windows and Linux
+
+For Windows, the steps here are all that are necesarry to enable secure boot.  Just follow the instructions in [Enable Guest Secure Boot](#enable-guest-secure-boot).
+
+For Linux, it is required to follow the Linux distribution's instructions
+for enabling UEFI Secure Boot by installing shim and the distribution's
+signed binaries.  After following those instructions from the distribution,
+follow the instructions in [Enable Guest Secure Boot](#enable-guest-secure-boot).
+
+#### Enable Guest Secure Boot
+
+#### Requirements
+* A UEFI guest VM.
+
+##### Download and Install Certificates
+
+Before enabling UEFI Secure Boot for guest VMs, first execute the `secureboot-certs` script.  This tool downloads, formats, and installs the publically available certificates from Microsoft.
+
+From the XCP-ng CLI:
+
+```
+# Download and install MS certificates
+secureboot-certs
+```
+
+:::tip
+Running `secureboot-certs` only needs to be done once per host, not per VM.
```

Comment:
Needs to be re-done after a host upgrade using the installation ISO, I suppose.

Reply:
Seems right to me, as the RPM declares `/usr/share/uefistored/` in its file section.

Question: could this be in the RPM's `%install` to eliminate this?  It hits the network, so maybe would be called with ` || true` to squelch the error on machines with no public net access on install.

Reply:
My point was rather than a host upgrade erases everything that is not in the XAPI (roughly). So you will probably lose the certs. You can test by upgrading a 8.2 to 8.2 again and see what happens.

About `%install` I think you mean `%post`, because `%install` is executed only in the build system. Should it be done in `%post`? Probably not because that wouldn't work during initial host installation nor host upgrades that aren't yum-style upgrades.

Reply:
Are host upgrades via the ISO not done via RPM?  Which then removes old `%files` files as it upgrades to the newer package?

In any case, yes.  These certs will be lost, so any new VMs will not have them after an upgrade until the script is called again.  The old VMs will still have them and their secure boot state will preserved.

Just to spread the knowledge, when a VM first boots `uefistored` loads the certs into the VMs XAPI DB entry.  uefistored will use those certs from XAPI for the lifetime of the VM, or until the user calls `varstore-sb-state <vm-uuid> setup` while the VM is shutdown, which clears the XAPI DB entry of the certs.

These certs (currently in `/usr/share/uefistored`) are the only non-temporary data used for guest secure boot.  Everything else is in the `chroot` that gets cleaned up by XAPI (like a socket file,  the chroot directory itself, etc...)... so nothing should be lost by system upgrade as long as the XAPI DB persists.

Reply:
Host upgrades copy  the entirety of your partition to a backup partition, make a fresh install, then restores what it must restore from the backup (XAPI db notably). So we need to inform users that they will have to run `secureboot-certs` again.

Reply:
About RPM, it never removes files that it doesn't own. Any files you added to a directory owned by the RPM stay. The directory is only removed if non-empty.

Reply:
Ah, thanks for the clarification on RPM only removing directories if non-empty.

Sounds good on adding that to the docs.